---
title: "First model"
output: html_notebook
---

So from the variables I looked at previously, some starting points:

* case ~ nesting habitat
    * Basically, NOGO roost in the same kind of places that they nest.
* case ~ canopy closure + basal area + density
    * Kitchen sink of promising variables BUT canopy and basal are highly correlated (0.8 correlation coefficient)
* case ~ canopy closure + vertical complexity + density
    * Alternate kitchen sink
* case ~ nesting habitat + canopy closure
    * Assumes the HSI is good but could be improved by an addition.
    
... and for all of them include site?

And I *think* use `glmer`.

```{r echo=TRUE, message=FALSE, warning=FALSE}
# Load some libraries.
library('ggplot2')
library('tidyverse')
library('lme4')

points <- read.csv('../data/interim/points.csv', stringsAsFactors=FALSE)

n.hab <- glmer(case ~ n_hab + (1|site), data=points,
      family=binomial(link='logit'))

sink.corr <- glmer(case ~ canopy.closure + basal.area + density + (1|site), data=points, family=binomial(link='logit'))
## Gives me weird error about 'Model is nearly unidentifiable'??

sink.alt <- glmer(case ~ canopy.closure + v.comp + density + (1|site), data=points, family=binomial(link='logit'))
## Same error.

n.plus <- glmer(case ~ n_hab + canopy.closure + (1|site), data=points, family=binomial(link='logit'))
```

None of those variables taken individually throws an error...

```{r echo=TRUE, message=FALSE, warning=FALSE}
canopy <- glmer(case ~ canopy.closure + (1|site), data=points, family=binomial(link='logit'))

density <- glmer(case ~ density + (1|site), data=points, family=binomial(link='logit'))

basal <- glmer(case ~ basal.area + (1|site), data=points, family=binomial(link='logit'))

vertical <- glmer(case ~ v.comp + (1|site), data=points, family=binomial(link='logit'))
```

So if I do a stupid thing and ignore the errors, I get this:

```{r echo=TRUE, message=FALSE, warning=FALSE}
AIC(n.hab, sink.corr, sink.alt, n.plus)

summary(n.hab)

summary(sink.corr)

summary(sink.alt)

summary(n.plus)
```
Now, `AIC()` also throws an error warning me that there are a different number of observations for each model. Which might be part of the "model unrecognizable" problem. I might need to sort through my data to select some lowest common denomenator of points.